packages/pangea-sdk/pangea/services/authn/authn.py

```diff
@@ -0,0 +1,196 @@
+# Copyright 2022 Pangea Cyber Corporation
+# Author: Pangea Cyber Corporation
+
+from typing import Optional
+
+from pangea.response import PangeaResponse
+from pangea.services.authn.models import (
+    IDProvider,
+    PasswordUpdateRequest,
+    PasswordUpdateResult,
+    Profile,
+    Scopes,
+    UserCreateRequest,
+    UserCreateResult,
+    UserDeleteRequest,
+    UserDeleteResult,
+    UserInviteDeleteRequest,
+    UserInviteDeleteResult,
+    UserInviteListResult,
+    UserInviteRequest,
+    UserInviteResult,
+    UserListRequest,
+    UserListResult,
+    UserLoginRequest,
+    UserLoginResult,
+    UserProfileGetRequest,
+    UserProfileGetResult,
+    UserProfileUpdateRequest,
+    UserProfileUpdateResult,
+)
+from pangea.services.base import ServiceBase
+
+
+class AuthN(ServiceBase):
+    """AuthN service client.
+
+    Provides methods to interact with the [Pangea AuthN Service](https://pangea.cloud/docs/api/authn).
+
+    The following information is needed:
+        PANGEA_TOKEN - service token which can be found on the Pangea User
+            Console at [https://console.pangea.cloud/project/tokens](https://console.pangea.cloud/project/tokens)
+
+    Examples:
+        import os
+
+        # Pangea SDK
+        from pangea.config import PangeaConfig
+        from pangea.services import AuthN
+
+        PANGEA_TOKEN = os.getenv("PANGEA_AUTHN_TOKEN")
+        authn_config = PangeaConfig(domain="pangea.cloud")
+
+        # Setup Pangea AuthN service
+        authn = AuthN(token=PANGEA_TOKEN, config=authn_config)
+    """
+
+    service_name: str = "authn"
+    version: str = "v1"
+
+    def __init__(
+        self,
+        token,
+        config=None,
+    ):
+        super().__init__(token, config)
+
+    #   - path: authn::/v1/password/update
+    # https://dev.pangea.cloud/docs/api/authn#change-a-users-password
+    def password_update(self, email: str, old_secret: str, new_secret: str) -> PangeaResponse[PasswordUpdateResult]:
+        input = PasswordUpdateRequest(email=email, old_secret=old_secret, new_secret=new_secret)
+        response = self.request.post("password/update", data=input.dict(exclude_none=True))
+        if response.raw_result is not None:
+            response.result = PasswordUpdateResult(**response.raw_result)
+        return response
+
+    #   - path: authn::/v1/user/create
+    # https://dev.pangea.cloud/docs/api/authn#create-user
+    def user_create(
```

Comment:
Should we have nest models for endpoints like this?

with the endpoint as user/create

Should the API be user.create ?

Reply:
@cj-f @ggallien13 solved this point, what do you think now? I was not sure about how much nested class should I use.